### PR TITLE
Add OW_UNION_DISABLE_CHECK and OW_FLAG_MOVE_UNION_ROOM_CHECK

### PR DESCRIPTION
**Pull request by pkmnsnfrn**
# Description
Player enters Pokemon Center without lag

* Adds `OW_UNION_DISABLE_CHECK`. When `TRUE`, the nurse does not inform the player if there is a trainer waiting in the Union Room. This speeds up the loading of the Pokémon Center.
* Adds `OW_FLAG_MOVE_UNION_ROOM_CHECK`. If this flag is set, the game will only check if players are in the Union Room while healing Pokémon, and not when players enter the Pokémon Center. This speeds up the loading of the Pokémon Center. This is ignored if `OW_UNION_DISABLE_CHECK` is `TRUE`.

#### `OW_UNION_DISABLE_CHECK`
When this is `TRUE`:
* `InitUnionRoom` will never run.
* `BufferUnionRoomPlayerName` will never run.

#### `OW_FLAG_MOVE_UNION_ROOM_CHECK`
When this is defined:
* `CableClub_OnResume` (meaning `InitUnionRoom`) will run after the Nurse sets the balls on the healing machine, instead of when the player enters the Pokémon Center.

# Testing
## Clean Branch
You can recreate this branch by applying a patch or pulling the repo. From a clean version of expansion's upcoming, you can either:

### Patch
`wget https://files.catbox.moe/o0qs9q.patch -O union.patch ; git apply union.patch ; rm union.patch`

### Repo
`git remote add psf-expansion https://github.com/PokemonSanFran/pokeemerald-expansion/ ; git pull psf-expansion noAdapterUnionCheck2`

## Manual Tests
After replicating the branch, to recreate my testing environment, you can either directly download the debug script, or manually create the changes.

#### `OW_UNION_DISABLE_CHECK` == `TRUE` && `OW_FLAG_MOVE_UNION_ROOM_CHECK ` == `0`
`wget https://files.catbox.moe/e2aojo.h -O include/config/overworld.h`
#### `OW_UNION_DISABLE_CHECK` == `FALSE` && `OW_FLAG_MOVE_UNION_ROOM_CHECK ` == `0`
`wget https://files.catbox.moe/qkdd2c.h -O include/config/overworld.h`
#### `OW_UNION_DISABLE_CHECK` == `TRUE` && `OW_FLAG_MOVE_UNION_ROOM_CHECK ` != `0`
`wget https://files.catbox.moe/semp2p.h -O include/config/overworld.h`
#### `OW_UNION_DISABLE_CHECK` == `FALSE` && `OW_FLAG_MOVE_UNION_ROOM_CHECK ` != `0`
`wget https://files.catbox.moe/xf1abp.h -O include/config/overworld.h`

### Manual Testing
* Modify `include/config/overworld.h` to the desired values
* Compile, start a new save
* Run Cheat Start
* Leave the truck
* Enter the Center

## Verified Scenarios
All videos start from a Player that has used Cheat Start and saved outside of the Pokemon Center.
* Walk into Pokemon Center
* Heal Pokemon with Nurse Joy
* Walk outside of Pokemon Center
* Walk into Pokemon Center

#### `OW_UNION_DISABLE_CHECK` == `TRUE` && `OW_FLAG_MOVE_UNION_ROOM_CHECK ` == `0`
https://github.com/user-attachments/assets/455627da-889f-4b9d-a001-f4e6ea1b413e
#### `OW_UNION_DISABLE_CHECK` == `FALSE` && `OW_FLAG_MOVE_UNION_ROOM_CHECK ` == `0`
https://github.com/user-attachments/assets/a7fb246f-969c-4702-a800-06d7720e3bd4
#### `OW_UNION_DISABLE_CHECK` == `TRUE` && `OW_FLAG_MOVE_UNION_ROOM_CHECK ` != `0`
https://github.com/user-attachments/assets/6b770574-a15a-49ab-96be-0302bdd0f23a
#### `OW_UNION_DISABLE_CHECK` == `FALSE` && `OW_FLAG_MOVE_UNION_ROOM_CHECK ` != `0`
https://github.com/user-attachments/assets/aea98be9-99ff-4830-b9dd-a54879109de5

# **People who collaborated with me in this PR**
This was originally [written by](<https://github.com/Deokishisu/CrystalDust/commit/7388c921a41962e042dda63f13d76372df22bc7f#diff-efd2003f4818c019a3cb52bd30da5d2eff9fb2d81b1a643819de13ec7b4b6196L7>) @Deokishisu. Give them all the credit.

# Features this PR does NOT handle:

# Discord Contact Info
I am `pkmnsnfrn` on Discord.

### CODE REVIEW

**Comment by pkmnsnfrn** [comment]
this is the improved version of https://github.com/rh-hideout/pokeemerald-expansion/pull/5050

File: data/scripts/cable_club.inc [hunk 1]
```diff
@@ -1258,7 +1258,12 @@ EventScript_CloseMossdeepGameCornerBarrier::
 
 CableClub_OnResume:
 	special InitUnionRoom
+.if OW_FLAG_MOVE_UNION_ROOM_CHECK != 0
+	return
+.else
 	end
+.endif
+
```
Comment (Bassoonian) [inline data/scripts/cable_club.inc:1266]: Redundant extra newline

File: data/scripts/pkmn_center_nurse.inc [hunk 2]
```diff
@@ -36,6 +36,13 @@ EventScript_PkmnCenterNurse_TakeAndHealPkmn::
 	applymovement VAR_0x800B, Movement_PkmnCenterNurse_Turn @ Changed from Common_Movement_WalkInPlaceFasterLeft to force the follower to enter their Poké Ball
 	waitmovement 0
 	dofieldeffect FLDEFF_POKECENTER_HEAL
+.if OW_UNION_DISABLE_CHECK == FALSE
+.if OW_FLAG_MOVE_UNION_ROOM_CHECK != 0
```
Comment (Bassoonian) [inline data/scripts/pkmn_center_nurse.inc:40]: I'm not sure, but is it possible to use `&&` in an `.if` statement? If so, that would clean this up considerably.

File: include/config/overworld.h [hunk 3]
```diff
@@ -107,5 +107,7 @@
 
 // Pokémon Center
 #define OW_IGNORE_EGGS_ON_HEAL     GEN_LATEST               // In Gen 4+, the nurse in the Pokémon Center does not heal Eggs on healing machine.
```
Comment (Bassoonian) [inline include/config/overworld.h:109]: I would indent `GEN_LATEST` to align with the new configs right below it

Comment (pkmnsnfrn) [reply include/config/overworld.h]: [`a8e9724` (#5448)](https://github.com/rh-hideout/pokeemerald-expansion/pull/5448/commits/a8e972429c4b86be20ed72da04107652f6cf4c88)

File: data/scripts/pkmn_center_nurse.inc (hunk 2 above)
Comment (pkmnsnfrn) [reply data/scripts/pkmn_center_nurse.inc]: [`0e9ea43` (#5448)](https://github.com/rh-hideout/pokeemerald-expansion/pull/5448/commits/0e9ea43a912e271ba87c1c6dccda761356a65812)

File: data/scripts/cable_club.inc (hunk 1 above)
Comment (pkmnsnfrn) [reply data/scripts/cable_club.inc:1266]: [`0e9ea43` (#5448)](https://github.com/rh-hideout/pokeemerald-expansion/pull/5448/commits/0e9ea43a912e271ba87c1c6dccda761356a65812)

File: include/config/overworld.h [hunk 4]
```diff
@@ -106,6 +106,8 @@
 #define OW_POPUP_BW_ALPHA_BLEND    FALSE                    // Enables alpha blending/transparency for the pop-ups. Mainly intended to be used with the black color option.
 
 // Pokémon Center
-#define OW_IGNORE_EGGS_ON_HEAL     GEN_LATEST               // In Gen 4+, the nurse in the Pokémon Center does not heal Eggs on healing machine.
+#define OW_IGNORE_EGGS_ON_HEAL           GEN_LATEST               // In Gen 4+, the nurse in the Pokémon Center does not heal Eggs on healing machine.
+#define OW_UNION_DISABLE_CHECK           FALSE              // When TRUE, the nurse does not inform the player if there is a trainer waiting in the Union Room. This speeds up the loading of the Pokémon Center.
```
Comment (Bassoonian) [inline include/config/overworld.h:110]: The comments are not lined out equally :)

Comment (pkmnsnfrn) [reply include/config/overworld.h]: [`d620914` (#5448)](https://github.com/rh-hideout/pokeemerald-expansion/pull/5448/commits/d620914599e8db359859ac26fe5c72203238b325)

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**